pkg/cable/libreswan/libreswan.go

```diff
@@ -359,6 +427,166 @@ func whack(args ...string) error {
 // ConnectToEndpoint establishes a connection to the given endpoint and returns a string
 // representation of the IP address of the target endpoint.
 func (i *libreswan) ConnectToEndpoint(endpointInfo *natdiscovery.NATEndpointInfo) (string, error) {
+	if i.authMode == AuthModeCert {
+		return i.connectToEndpointCertMode(endpointInfo)
+	}
+
+	return i.connectToEndpointPSKMode(endpointInfo)
+}
+
+func appendConnectionStanza(confPath, stanza, connName string) error {
+	if err := removeConnectionStanza(confPath, connName); err != nil {
+		return err
+	}
+
+	f, err := os.OpenFile(confPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+	if err != nil {
+		return errors.Wrap(err, "error opening file")
+	}
+
+	defer f.Close()
+
+	if !strings.HasSuffix(stanza, "\n") {
+		stanza += "\n"
+	}
+
+	_, err = f.WriteString(stanza)
+
+	return errors.Wrap(err, "error writing to file")
+}
+
+func removeConnectionStanza(confPath, connName string) error {
+	data, err := os.ReadFile(confPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+
+		return errors.Wrap(err, "error reading file")
+	}
+
+	lines := strings.Split(string(data), "\n")
+	var out []string
+	inStanza := false
+
+	for _, line := range lines {
+		if strings.HasPrefix(line, "conn ") && strings.TrimSpace(line) == ("conn "+connName) {
```

Comment:
It doesn't seem like `HasPrefix` is needed here, also if it started with a space `HasPrefix` would be false.


```suggestion
		if strings.TrimSpace(line) == ("conn "+connName) {
```